tests/steinberger_latent_heat.prm

```diff
@@ -0,0 +1,95 @@
+# This is a copy of the steinberger compressible test. However, it activates
+# latent heat by computing the thermal expansivity and specific heat as
+# derivatives of the enthalpy, which leads to temperature jumps across phase
+# transitions.
```

Comment:
Have you verified that the temperature jumps across the phase transitions make sense? I would have taken the test we have for the grain size material model and copied that.